File: RiotGames/Dto/Match/PerkStyleDto.cs
using Newtonsoft.Json;

namespace potg.RiotGames.Dto.Match;

public record PerkStyleDto
{
    [JsonProperty("description")]
    public string Description { get; init; }

    [JsonProperty("selections")]
    public IReadOnlyCollection<PerkStyleSelectionDto> Selections { get; init; }

    [JsonProperty("style")]
    public long StyleStyle { get; init; }
}

File: src/models/dto/riot/lol/match/perkStyle.dto.ts
import { Field, ObjectType } from "@nestjs/graphql";
import { PerkStyleSelectionDto } from "./perkStyleSelection.dto";

@ObjectType()
export class PerkStyleDto {
  @Field()
  description: string;
  @Field((type) => [PerkStyleSelectionDto])
  selections: PerkStyleSelectionDto[];
  @Field()
  style: number;
}
